src/Microsoft.Restier.Publishers.OData/Model/RestierModelBuilder.cs

```diff
@@ -49,7 +55,19 @@ class RestierModelBuilder : IModelBuilder
 
             context.EntitySetTypeMapCollection.Clear();
 
+            ExtendModel(builder);
+
             return builder.GetEdmModel();
         }
+
+        /// <summary>
+        /// This method allow user to extend the model with more entity type and entity set before operation model builder starts
+        /// A sub class need to be implemented and register as DI service like "AddTransient&lt;RestierModelBuilder, CustomizedRestierModelBuilder&gt;"
+        /// </summary>
+        /// <param name="builder"></param>
+        public virtual void ExtendModel(ODataConventionModelBuilder builder)
```

Comment:
@rayao we already has model builder as chain service, but it does not support a case very well when we make EF entity type and set model build in OData publisher which is user need to add more entity type and set before operation model builder starts, I assume this is for very small group of consumers but we do see consumers need this now. 
We have several options for consumers,
1. Without this method, user can still do this, but user need to overwrite all logic in RestierModelExtender and add his own logic
2. Add a new virtual method to allow user add new entity set and entity type before operation model build start.
3. Introduce a new interface and another chain of service as you proposed.
I was thinking the third option is too complex and even normal user will confuse with IModelBuilder and IModelExtender.


We will re-think this when we resolve issue #447, we may need to move operation model builder from ODL to Web Api model builder too.